src/payment/strategies/googlepay/googlepay-payment-strategy.ts

```diff
@@ -157,6 +141,26 @@ export default class GooglePayPaymentStrategy implements PaymentStrategy {
         };
     }
 
+    private _getGooglePayOptions(options: PaymentInitializeOptions): GooglePayPaymentInitializeOptions {
```

Comment:
I think the whole idea of having `GooglePayPaymentProcessor` and provider-specific initializers like `GooglePayCheckoutcomInitializer` implementing a shared `GooglePayInitializer` interface is to avoid inheritance. You can have a different implementation for each provider. But because these implementations share the same interface, you can inject them into the same strategy implementation. I'd try to follow that pattern instead of using inheritance for this particular provider so that it is consistent with other providers like Authorize.net and Braintree.

Reply:
@davidchin the added step i need for this provider couldn't be done on the initializer, it could have been on the parseResponse that return the token from google pay parsed, as i need to transform it before sending it to bigpay, but that method its not tailored to be async, i would have needed to modify its template and every other provider implementing googlepay, i was advised not too, so ended up extending the strategy and adding the step i needed